### ISSUE TITLE
MutableSelectionSet child objects missing selections

### ISSUE THREAD

**Issue by pixelmatrix** [apollographql/apollo-ios#3437]
### Summary

I have a mutable fragment that has relationships to other objects, like this:

```gql
fragment MutableIssueFields on Issue @apollo_client_ios_localCacheMutation {
  id
  state {
    id
    name
  }
}
```

Codegen ends up creating `MutableIssueFields` and `MutableIssueFields.State`. The parent object `MutableIssueFields` has a` __selections` property as you'd expect, but `MutableIssueFields.State` does not.

I'm trying to change the value from one "state" to another using a local cache mutation. However, when I try to load the `MutableIssueFields.State` object from the cache, I get an empty object because the `__selections` field is empty.

I'm calling it like this:

```swift
store.withinReadWriteTransaction { tx in
  let newState = try tx.readObject(ofType: MutableIssueFields.State.self, withKey: "State:\(id)")
  try tx.updateObject(ofType: MutableIssueFields.self) { obj in
    obj.state = newState
  }
}
```

Is this something that should be supported? If not, is there a preferred way to update this relationship to another existing object in the cache?

### Version

1.14.1

### Steps to reproduce the behavior

1. Create a fragment marked with @apollo_client_ios_localCacheMutation, that references some child object
2. Populate cache with data that will fulfill the fragment
3. Use a read write transaction to try to fetch the child object using `readObject`
4. Observe the child object is returned with empty data

**Comment by AnthonyMDev** [comment]
No your code looks correct, and this should be supported. It seems like it's a bug that the `MutableIssueFields.State` has an empty `__selections`. That shouldn't be.

We'll need to see if we can reproduce this.

**Comment by calvincestari** [comment]
@pixelmatrix can you please share the generated code for `MutableIssueFields` so I can take a look at the whole thing.

Attached is my attempted reproduction case but I get a populated `__selections` property in `MutableIssueFields.State` out of codegen. I didn't try the actual cache update because I have no reason to believe it wouldn't work with `__selections` being correctly generated. If you can manipulate the reproduction sample so that it generates the same output as you get that would certainly help to debug. Thanks.

[issue3437.zip](https://github.com/user-attachments/files/18806793/issue3437.zip)

**Comment by pixelmatrix** [comment]
Here's what I have:

[MutableIssueFields.graphql.swift.zip](https://github.com/user-attachments/files/18806880/MutableIssueFields.graphql.swift.zip)

**Comment by calvincestari** [comment]
@pixelmatrix I think I've managed to replicate the missing `_selections` in the `WorkflowState` fragment with the 1.14.1 version you specified. That behaviour changed from 1.15.0 though and I can no longer replicate it with any version since then.

Have you executed codegen with a version 1.15.0, or greater, to see if it resolves the issue?

**Comment by pixelmatrix** [comment]
We're using Apollo 1.17.0 currently, which I assumed includes codegen too, but maybe it's not getting updated automatically in my setup. I'll dig into this more and report back.

**Comment by calvincestari** [comment]
If you're executing codegen using the CLI then it is included with apollo-ios as we distribute a codegen binary with each version. However, if you're generating through code and depending on the apollo-ios-codegen package directly then you will need to be updating that dependency too.

**Comment by pixelmatrix** [comment]
It's done via the CLI, but for some reason my `apollo-ios-codegen` is a binary and not a link as the docs suggest. I set it up by running the Xcode Install CLI action a while back. Do I need to be running that action every time I update the SDK as well?

**Comment by calvincestari** [comment]
You can check the version of the binary with the command `./apollo-ios-cli --version`.

**Comment by pixelmatrix** [comment]
Okay, I updated codegen to match our current version of the library and while it did make a ton of changes to the generated models, I still don't see `__selections` for `MutableIssueFields`.

<img width="798" alt="Image" src="https://github.com/user-attachments/assets/8a4a49b5-0e1b-43a7-87de-39a7573d1833" />

**Comment by calvincestari** [comment]
Can you share your fragment definitions for this please. Are they defined exactly like you have in the description or is it multiple fragments within each other?

This is what gave me a shape similar to the MutableIssueFields.graphql.swift file you [shared last week](https://github.com/apollographql/apollo-ios/issues/3437#issuecomment-2660565359):
```graphql
fragment MutableIssueFields on Issue @apollo_client_ios_localCacheMutation {
  ...IssueDetailFields
}

fragment IssueDetailFields on Issue {
  id
  state {
    ...WorkflowStateFields
  }
}

fragment WorkflowStateFields on WorkflowState {
  id
  name
}

```

**Comment by pixelmatrix** [comment]
Yeah, it's actually composed of many fragments. The shape is essentially what you described above, but it goes through one more fragment (`partialIssueFields`):

```
fragment issueDetailFields on Issue {
  id
  # etc...
  ...partialIssueFields
}

fragment partialIssueFields on Issue {
  id
  # etc...
  state {
    ...workflowStateFields
  }

fragment workflowStateFields on WorkflowState {
  id
  type
  name
  color
  position
  description
  team {
    id
  }
}

fragment MutableIssueFields on Issue @apollo_client_ios_localCacheMutation {
  ...issueDetailFields
}
```

**Comment by calvincestari** [comment]
Ok, that'll help. I'll have another go at this in the morning.
